src/main/java/com/oltpbenchmark/benchmarks/templated/procedures/GenericQuery.java

```diff
@@ -48,33 +58,96 @@ public void run(Connection conn, List<Object> params) throws SQLException {
             while (rs.next()) {
-                //do nothing
+                // do nothing
             }
         }
         conn.commit();
     }
 
-    public PreparedStatement getStatement(Connection conn, List<Object> params) throws SQLException {
+    public PreparedStatement getStatement(Connection conn, List<Object> params,
+            HashMap<Integer, Object> randomGenerators) throws SQLException {
         QueryTemplateInfo queryTemplateInfo = this.getQueryTemplateInfo();
 
         PreparedStatement stmt = this.getPreparedStatement(conn, queryTemplateInfo.getQuery());
         String[] paramsTypes = queryTemplateInfo.getParamsTypes();
+        int j = 0;
         for (int i = 0; i < paramsTypes.length; i++) {
             if (paramsTypes[i].equalsIgnoreCase("NULL")) {
                 stmt.setNull(i + 1, Types.NULL);
+            } else if (paramsTypes[i].equalsIgnoreCase("RANDOM")) {
+                int index = i + j;
+                String distType = params.get(index).toString();
+                int minI, maxI;
+                switch (distType) {
+                    case "zipf":
+                        ZipfianGenerator zipfGen = (ZipfianGenerator) randomGenerators.get(index);
+                        stmt.setInt(i + 1, zipfGen.nextInt());
+                        j += 2;
+                        break;
+                    case "uniform":
+                        minI = Integer.parseInt(params.get(index + 1).toString());
+                        maxI = Integer.parseInt(params.get(index + 2).toString());
+                        Random rnd = (Random) randomGenerators.get(index);
+                        int uVal = rnd.nextInt(maxI - minI) + minI;
+                        stmt.setInt(i + 1, uVal);
+                        j += 2;
+                        break;
+                    case "binomial":
+                        minI = Integer.parseInt(params.get(index + 1).toString());
+                        maxI = Integer.parseInt(params.get(index + 2).toString());
+                        Random rng = (Random) randomGenerators.get(index);
+                        int bVal;
+                        do {
+                            bVal = (int) (minI + Math.abs(rng.nextGaussian()) * maxI);
+                        } while (bVal > maxI || bVal < minI);
+
+                        stmt.setInt(i + 1, bVal);
+                        j += 2;
+                        break;
+                    case "scrambled":
+                        ScrambledZipfianGenerator scramZipf = (ScrambledZipfianGenerator) randomGenerators.get(index);
+                        stmt.setInt(i + 1, scramZipf.nextInt());
+                        j += 2;
+                        break;
+                    case "string":
+                        maxI = Integer.parseInt(params.get(index + 1).toString());
+                        Random rand = (Random) randomGenerators.get(index);
+                        String randText = TextGenerator.randomStr(rand, maxI);
```

Comment:
This would only support uniform distribution for strings, correct?
Wouldn't it be more useful to specify 
a) the type of the param (string, int, float), and
b) the distribution (and parameters) of that type separately?